exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -149,11 +169,29 @@ func (r *Reconciler) reconcileDelete(_ context.Context, extensionConfig *runtime
 	return ctrl.Result{}, nil
 }
 
+func (r *Reconciler) secretToExtensionConfig(object client.Object) []reconcile.Request {
+	result := []ctrl.Request{}
+
+	extensionConfigs := runtimev1.ExtensionConfigList{}
+	if err := r.Client.List(context.TODO(), &extensionConfigs); err != nil {
+		return nil
+	}
+
+	for _, ext := range extensionConfigs.Items {
+		if _, ok := ext.GetAnnotations()[caBundleInjectFromSecretAnnotation]; ok {
+			name := client.ObjectKey{Namespace: ext.GetNamespace(), Name: ext.GetName()}
```

Comment:
we should only add the ExtensionConfigs to result which are using object, not all that have the annotation